influxql/ast.go

```diff
@@ -968,6 +994,53 @@ func (s *SelectStatement) validateDistinct() error {
 	return nil
 }
 
+func (s *SelectStatement) HasCountDistinct() bool {
+	for _, f := range s.Fields {
+		if c, ok := f.Expr.(*Call); ok {
+			if c.Name == "count" {
+				for _, a := range c.Args {
+					if _, ok := a.(*Distinct); ok {
+						return true
+					}
+				}
+			}
+		}
+	}
+	return false
+}
```

Comment:
Needs to check for `distinct(value)` also.  Line 1107 in the integration tests needs the FROM clause to catch this...
```
{s: `select count(distinct(too, many, arguments)) FROM myseries`, err: `count(distinct <field>) can only have one argument`},
```